### ISSUE TITLE
Tests segfault on MacOS

### ISSUE THREAD

**Issue by BastianZim** [pyscf/pyscf#1146]
This was originally reported in #1110 but I wanted to create a separate issue here.

Logs: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=432593&view=logs&j=e35d4f76-8ff2-5536-d795-df91e63eb9f7&t=fa7b4b17-b6ff-5c9c-8cfc-15f888c92310&l=5695

Output: 
```zsh
test_dhf (test_df_jk.KnownValues) ... .github/workflows/run_tests.sh: line 39: 12220 Segmentation fault: 11  nosetests pyscf/ -v --with-timer --with-cov --cov-report xml --cov-report term --cov-config .coveragerc --cov pyscf --exclude-dir=examples --exclude-dir=pyscf/pbc/grad -e test_bz -e h2o_vdz -e test_mc2step_4o4e -e test_ks_noimport -e test_jk_hermi0 -e test_j_kpts -e test_k_kpts -e high_cost -e skip -e call_in_background -e libxc_cam_beta_bug -e test_finite_diff_rks_eph -e test_finite_diff_uks_eph -e test_pipek -I test_kuccsd_supercell_vs_kpts\.py -I test_kccsd_ghf\.py -I test_h_.*\.py -I test_P_uadc_ea.py -I test_P_uadc_ip.py --exclude-test=pyscf/pbc/gw/test/test_kgw_slow_supercell.DiamondTestSupercell3 --exclude-test=pyscf/pbc/gw/test/test_kgw_slow_supercell.DiamondKSTestSupercell3 --exclude-test=pyscf/pbc/gw/test/test_kgw_slow.DiamondTestSupercell3 --exclude-test=pyscf/pbc/gw/test/test_kgw_slow.DiamondKSTestSupercell3 --exclude-test=pyscf/pbc/tdscf/test/test_krhf_slow_supercell.DiamondTestSupercell3 --exclude-test=pyscf/pbc/tdscf/test/test_kproxy_hf.DiamondTestSupercell3 --exclude-test=pyscf/pbc/tdscf/test/test_kproxy_ks.DiamondTestSupercell3 --exclude-test=pyscf/pbc/tdscf/test/test_kproxy_supercell_hf.DiamondTestSupercell3 --exclude-test=pyscf/pbc/tdscf/test/test_kproxy_supercell_ks.DiamondTestSupercell3 -I .*_slow.*py -I .*_kproxy_.*py -I test_proxy.py
/Users/runner/mambaforge/conda-bld/pyscf_1640435032163/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/python3.9/multiprocessing/resource_tracker.py:216: UserWarning: resource_tracker: There appear to be 4 leaked semaphore objects to clean up at shutdown
  warnings.warn('resource_tracker: There appear to be %d '
```

Build commands:
```yaml
- export CMAKE_CONFIGURE_ARGS="${CMAKE_ARGS} -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=$PREFIX -DCMAKE_INSTALL_LIBDIR=lib -DBUILD_SHARED_LIBS=ON -DBUILD_LIBCINT=OFF -DBUILD_LIBXC=OFF -DBUILD_XCFUN=OFF -LAH"
- python -m pip install . -vv
```

Environment: Azure pipelines

PR: https://github.com/conda-forge/staged-recipes/pull/17080

It happens every time the tests run.

**Comment by sunqm** [comment]
The segfault error was observed for a while also in the github actions. It might be related to issue #1140 . The fixes in PR #1142 (commit d096d6f00e92cab17e6574488049cded16488311) solved one source of segfault. Does the conda forge build points to the versions older than the commit d096d6f00e92cab17e6574488049cded16488311?

**Comment by BastianZim** [comment]
The conda-forge build points to the latest release which is `2.0.1`. I have added #1142 as a patch though so that is included. 

I'm still running into the `undefined symbol: AO2MOmmm_ket_nr_s2` error though on linux (Ref: https://github.com/pyscf/pyscf/issues/1118#issuecomment-1001024058) might that be connected?

I have not dug deep enough into the code to definitively say if this is the error but #1142 changed `_ao2mo.libao2mo.AO2MOtranse2_nr_s1` whereas the error is `undefined symbol: AO2MOmmm_ket_nr_s2`(1 vs 2 at the end), might that be the cause?

**Comment by BastianZim** [comment]
@sunqm Just checking if you maybe have an idea about what could cause this?

**Comment by BastianZim** [comment]
Note: This is in a different section than the current failure in the CI here but the same segfault: https://github.com/pyscf/pyscf/runs/5177244700?check_suite_focus=true#step:4:2578

**Comment by sunqm** [comment]
Fixed in pyscf v2.1
